tests/suite/graceful_recovery_test.go

```diff
@@ -256,10 +342,43 @@ func expectRequestToFail(appURL, address string) error {
 	return nil
 }
 
+func checkNGFFunctionality(teaURL, coffeeURL, ngfPodName string, files []string, ns *core.Namespace) {
+	Eventually(
+		func() error {
+			return checkForWorkingTraffic(teaURL, coffeeURL)
+		}).
+		WithTimeout(timeoutConfig.RequestTimeout).
+		WithPolling(500 * time.Millisecond).
+		Should(Succeed())
+
+	Expect(resourceManager.DeleteFromFiles(files, ns.Name)).To(Succeed())
+
+	Eventually(
+		func() error {
+			return checkForFailingTraffic(teaURL, coffeeURL)
+		}).
+		WithTimeout(timeoutConfig.RequestTimeout).
+		WithPolling(500 * time.Millisecond).
+		Should(Succeed())
+
+	Expect(resourceManager.ApplyFromFiles(files, ns.Name)).To(Succeed())
+	Expect(resourceManager.WaitForAppsToBeReadyWithPodCount(ns.Name, 2)).To(Succeed())
+
+	Eventually(
+		func() error {
+			return checkForWorkingTraffic(teaURL, coffeeURL)
+		}).
+		WithTimeout(timeoutConfig.RequestTimeout).
+		WithPolling(500 * time.Millisecond).
+		Should(Succeed())
+
+	checkContainerLogsForErrors(ngfPodName)
+}
+
 // checkContainerLogsForErrors checks both nginx and ngf container's logs for any possible errors.
-// Since this function retrieves all the logs from both containers and the NGF pod may be shared between tests,
-// the logs retrieved may contain log messages from previous tests, thus any errors in the logs from previous tests
-// may cause an interference with this test and cause this test to fail.
+// Since this function retrieves all the logs from both containers and the NGF pod is not torn down between
+// other graceful-recovery tests, the logs retrieved may contain log messages from previous tests, thus any errors in
+// the logs from previous tests may cause cascade and cause following graceful-recovery tests to error.
```

Comment:
Can we call [AbortSuite](https://pkg.go.dev/github.com/onsi/ginkgo/v2#AbortSuite) to stop the graceful recovery test suite after a failure?

Reply:
I think the way it is currently is fine especially since with the new tests I just added, if there is a failure it would most definitely cascade downwards to the other tests also. So I would need to weave AbortSuite throughout the entire test suite which I think is a bit unnecessary.

Reply:
Ok, do we need to add a FIXME for this? IMO one test failure shouldn't cause other tests to fail. We should either stop execution, or tear down NGF in between the tests so each test is independent and failures do not cascade.

Reply:
I decided to tear down NGF in between the tests so each test should be independent and failures do not cascade. One potential missed benefit is that gain of running multiple of these failures on the same NGF instance (So we know that a single deployment of NGF can recover from all 4 of these situations), but I think that is okay.